src/translate-terms.js

```diff
@@ -269,3 +183,143 @@ export default function () {
 
 }
 
+export function fieldHasValidSubfield0(field, defaultLex = undefined) {
+  const lex = defaultLex || mapTagToLex(field.tag);
+  return field.subfields.some(sf => isValidSubfield0(sf, lex));
+}
+
+export function isValidSubfield0(subfield, lex = '???') {
+  if (subfield.code !== '0') {
+    return false;
+  }
+  // Should we support https as well?
+  if (['yso', 'yso/fin', 'yso/swe'].includes(lex) && subfield.value.match(/^http:\/\/www\.yso\.fi\/onto\/yso\/p[0-9]+$/u)) {
+    return true;
+  }
+  if (['slm', 'slm/fin', 'slm/swe'].includes(lex) && subfield.value.match(/^http:\/\/urn\.fi\/URN:NBN:fi:au:slm:s[0-9]+$/u)) {
+    return true;
+  }
+  return false;
+}
+
+
+export function buildUri(lex, id) {
+  if (['yso', 'yso/fin', 'yso/swe'].includes(lex)) {
+    //return `http%3A%2F%2Fwww.yso.fi%2Fonto%2Fyso%2Fp${id}`;
+    return `http://www.yso.fi/onto/yso/p${id}`;
+  }
+  if (['slm', 'slm/fin', 'slm/swe'].includes(lex)) {
+    return `http://urn.fi/URN:NBN:fi:au:slm:s${id}`;
+  }
+  return undefined;
+}
+
+function mapTagToLex(tag) {
+  if (tag === '655') {
+    return 'slm';
+  }
+  return 'yso';
+}
+
+export async function getTermData(uri) {
+  nvdebug(`getTermData(${uri})`);
+  if (termCache[uri]) { // Don't think current implementation uses the cache any more.
```

Comment:
translate-terms.js row 12 has termCache defined.
does this work (for drop-terms, if termCache is undefined there?)